rancher/rancher
# Remove `omitempty` tags to ensure fields persist across json marshals

Issue: https://github.com/rancher/rancher/issues/37597
 
# Problem
 UI defaults were incorrectly being applied to fields which were equal to a zero value (such as `false`). This resulted in the loss of some initial configurations of the Upgrade Strategy after a subsequent edit of the Configuration. The UI was applying its defaults to fields which it could not find in the `GET` request made to the backend upon attempting to edit the Configuration. 

# Solution
These fields were being dropped by the web-hook upon marshaling due to the `omitempty` tag on the struct `DrainOptions`. This caused those fields to be missing in the subsequent `GET` request made by the UI. This can be avoided by removing the `omitempty` tags.
 
# Testing
 + Begin creating an RKE2 cluster on any infrastructure-provider
 + Under `Cluster Configuration` find the `Upgrade Strategy` tab.
 + Select `yes` for `Drain Nodes` for the `Control Plane` 
 + deselect the `Delete pods using emptyDir volumes`
 + Create the cluster
 + On the Cluster Management screen, select the cluster and click on `edit config`
 + navigate to the `Upgrade Strategy` field and observe that the `Delete pods using emptyDir volumes` box has been not been re-activated, even though it is equal to that fields zero value.
